### PR TITLE
includeconfigsource: add tests for watch file and template files

### PR DESCRIPTION
**Pull request by ccordi**
This PR fixes a bug in the retrieve function when using templated files. 

It also adds unit tests to cover templated and watched files.

### CODE REVIEW

File: internal/configsource/includeconfigsource/source.go [hunk 1]
```diff
@@ -60,7 +60,13 @@ func (is *includeConfigSource) Retrieve(_ context.Context, selector string, para
 	}
 
 	var buf bytes.Buffer
-	if err = tmpl.Execute(&buf, paramsConfigMap); err != nil {
+	var params map[string]interface{}
```
Comment (rmfitzpatrick) [inline internal/configsource/includeconfigsource/source.go:63]: nit, we've moved to `any` but don't have a lint check for it atm.

**Comment by rmfitzpatrick** [comment]
Do you mind adding at least one integration test similar to https://github.com/signalfx/splunk-otel-collector/blob/main/tests/general/envvar_expansion_test.go#L83 that expands yaml ~mappings from multiple files (maybe in a new `general/config` suite? It'd be helpful to vet the config source complying with the actual service config resolution system with scalar and blobs: https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/main.go#L77

File: tests/general/template_config_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package tests
```
Comment (rmfitzpatrick) [inline tests/general/template_config_test.go:1]: mind renaming the file to something like `include_config_source_test.go`?

**Review by atoulme — APPROVED** [review]

File: tests/general/include_config_source_test.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		Build()
+
+	require.NotNil(t, collector)
+	require.NoError(t, err)
+
+	err = collector.Start()
+	require.NoError(t, err)
+
+	_, ok := collector.(*testutils.CollectorProcess)
+	require.True(t, ok)
```
Comment (rmfitzpatrick) [inline tests/general/include_config_source_test.go:36]:
```suggestion
```

I don't think these are necessary when not using the `CollectorProcess` directly.

**Review by rmfitzpatrick — APPROVED** [review]

File: tests/general/template_config_test.go (hunk 2 above)
Comment (ccordi) [reply tests/general/template_config_test.go]: done

**Pull request merged**